# Predict Breast Cancer

Predict the class of breast cancer (malignant or ‘bad’ versus benign or ‘good’) from the features of images taken from breast samples. Ten biological attributes of the cancer cell nuclei have been calculated from the images

##### Import packages

In [34]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

##### Examine the data

In [50]:
#reading the data
cancer = pd.read_csv('data/cancer.data', header=None, index_col=0)
cancer.head()

1   2   3   4   5   6   7   8   9   10
0                                              
1000025   5   1   1   1   2   1   3   1   1   2
1002945   5   4   4   5   7  10   3   2   1   2
1015425   3   1   1   1   2   2   3   1   1   2
1016277   6   8   8   1   3   4   3   7   1   2
1017023   4   1   1   3   2   1   3   1   1   2

In [51]:
#renaming columns
cancer.rename(columns = {1:'Clump Thickness', 2:'Uniformity of Cell Size',
                        3:'Uniformity of Cell Shape', 4:'Marginal Adhesion',
                        5:'Single Epithelial Cell Size',6:'Bare Nuclei',
                        7:'Bland Chromatin',8:'Normal Nucleoli',
                        9:'Mitoses', 10:'Class'}, inplace=True) 
cancer.head()

Clump Thickness  Uniformity of Cell Size  Uniformity of Cell Shape  \
0                                                                             
1000025                5                        1                         1   
1002945                5                        4                         4   
1015425                3                        1                         1   
1016277                6                        8                         8   
1017023                4                        1                         1   

         Marginal Adhesion  Single Epithelial Cell Size Bare Nuclei  \
0                                                                     
1000025                  1                            2           1   
1002945                  5                            7          10   
1015425                  1                            2           2   
1016277                  1                            3           4   
1017023                  3                            2           1   

         Bland Chromatin  Normal Nucleoli  Mitoses  Class  
0                                                          
1000025                3                1        1      2  
1002945                3                2        1      2  
1015425                3                1        1      2  
1016277                3                7        1      2  
1017023                3                1        1      2

#### Inspecting data

In [52]:
#DataFrame information
cancer.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 699 entries, 1000025 to 897471
Data columns (total 10 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   Clump Thickness              699 non-null    int64 
 1   Uniformity of Cell Size      699 non-null    int64 
 2   Uniformity of Cell Shape     699 non-null    int64 
 3   Marginal Adhesion            699 non-null    int64 
 4   Single Epithelial Cell Size  699 non-null    int64 
 5   Bare Nuclei                  699 non-null    object
 6   Bland Chromatin              699 non-null    int64 
 7   Normal Nucleoli              699 non-null    int64 
 8   Mitoses                      699 non-null    int64 
 9   Class                        699 non-null    int64 
dtypes: int64(9), object(1)
memory usage: 60.1+ KB


From the information above we see that we have;
- 10 columns
- 699 entries
- 1 object dtype and 9 int64

##### Bare Nuclei 
Bare Nuclei is an object instead of integer.

In [53]:
#inspect the missing value
cancer['Bare Nuclei'].value_counts()

1     402
10    132
2      30
5      30
3      28
8      21
4      19
?      16
9       9
7       8
6       4
Name: Bare Nuclei, dtype: int64

The dataset has missing value it is spotted on 16 '?'.

In [54]:
mode = cancer['Bare Nuclei'].value_counts().index[0]
# Replacing the '?'s with common digit 
cancer['Bare Nuclei'] = cancer['Bare Nuclei'].replace('?', mode).astype('int64') #converting object to integer
cancer['Bare Nuclei'].value_counts()

1     418
10    132
5      30
2      30
3      28
8      21
4      19
9       9
7       8
6       4
Name: Bare Nuclei, dtype: int64

I have replaced the missing values with the most occuring digit, in this instance which is 1. Therefore the value that was in the missing column I have added in the most occuring number. Thereafter since all the values would be numbers, I change the column type from object to integer.

##### Descriptive statistic

In [55]:
cancer.describe()

Clump Thickness  Uniformity of Cell Size  Uniformity of Cell Shape  \
count       699.000000               699.000000                699.000000   
mean          4.417740                 3.134478                  3.207439   
std           2.815741                 3.051459                  2.971913   
min           1.000000                 1.000000                  1.000000   
25%           2.000000                 1.000000                  1.000000   
50%           4.000000                 1.000000                  1.000000   
75%           6.000000                 5.000000                  5.000000   
max          10.000000                10.000000                 10.000000   

       Marginal Adhesion  Single Epithelial Cell Size  Bare Nuclei  \
count         699.000000                   699.000000   699.000000   
mean            2.806867                     3.216023     3.486409   
std             2.855379                     2.214300     3.621929   
min             1.000000                     1.000000     1.000000   
25%             1.000000                     2.000000     1.000000   
50%             1.000000                     2.000000     1.000000   
75%             4.000000                     4.000000     5.000000   
max            10.000000                    10.000000    10.000000   

       Bland Chromatin  Normal Nucleoli     Mitoses       Class  
count       699.000000       699.000000  699.000000  699.000000  
mean          3.437768         2.866953    1.589413    2.689557  
std           2.438364         3.053634    1.715078    0.951273  
min           1.000000         1.000000    1.000000    2.000000  
25%           2.000000         1.000000    1.000000    2.000000  
50%           3.000000         1.000000    1.000000    2.000000  
75%           5.000000         4.000000    1.000000    4.000000  
max          10.000000        10.000000   10.000000    4.000000